TabloidCLI/UserInterfaceManagers/PostDetailManager.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System;
+using System.Collections.Generic;
+using TabloidCLI.Models;
+using TabloidCLI.Repositories;
+
+namespace TabloidCLI.UserInterfaceManagers
+{
+    internal class PostDetailManager : IUserInterfaceManager
+    {
+        private IUserInterfaceManager _parentUI;
+        private AuthorRepository _authorRepository;
+        private PostRepository _postRepository;
+        private TagRepository _tagRepository;
+        private int _postId;
+
+        public PostDetailManager(IUserInterfaceManager parentUI, string connectionString, int postId)
+        {
+            _parentUI = parentUI;
+            _authorRepository = new AuthorRepository(connectionString);
+            _postRepository = new PostRepository(connectionString);
+            _tagRepository = new TagRepository(connectionString);
+            _postId = postId;
+        }
+
+        public IUserInterfaceManager Execute()
+        {
+            Post post = _postRepository.Get(_postId);
+            Console.WriteLine($"{post.Title} Details");
+            Console.WriteLine(" 1) View");
+            Console.WriteLine(" 2) View Blog Posts");
+            Console.WriteLine(" 3) Add Tag");
+            Console.WriteLine(" 4) Remove Tag");
+            Console.WriteLine(" 0) Go Back");
+
+            Console.Write("> ");
+            string choice = Console.ReadLine();
+            switch (choice)
+            {
+                case "1":
+                    View();
+                    return this;
+                case "2":
+                    ViewBlogPosts();
+                    return this;
+                case "3":
+                    //AddTag();
+                    return this;
+                case "4":
+                    //RemoveTag();
+                    return this;
+                case "0":
+                    return _parentUI;
+                default:
+                    Console.WriteLine("Invalid Selection");
+                    return this;
+            }
+        }
+
+        private void View()
+        {
+            Post post = _postRepository.Get(_postId);
+            Console.WriteLine($"Title: {post.Title}");
+            Console.WriteLine($"Url: {post.Url}");
+            Console.WriteLine($"Date:{post.PublishDateTime}");
+        }
+
+        private void ViewBlogPosts()
+        {
+            //List<Post> posts = _postRepository.GetByAuthor(_authorId);
```

Comment:
yeah, here again, this code is from the other repo, if you need it later, you can get it from there.